JSMapr.js

```diff
@@ -270,6 +270,53 @@
 		return srcObj;
 	}
 
+	function opAddToArray(srcObj, loc, elemToAdd) {
+		// if not valid location, abort
+		if (!isLocValid(loc)) return srcObj;
+
+		var val = getObjectAtLoc(srcObj, loc);
+		if (val === undefined || val === null) {
+			// nothing there, add as single element array
+			srcObj = setObjectAtLoc(srcObj, loc, [ elemToAdd ]);
+		} else if (isArray(val)) {
+			// already an array, push new element
+			val.push(elemToAdd);
+		} else if (isObject(val)) {
+			// this is an object, do nothing
+		} else {
+			// something else, do nothing
+		}
+		return srcObj;
+	}
+
+	function opMoveToArray(srcObj, srcLoc, destLoc) {
+		// if not valid locations, abort
+		if (!isLocValid(srcLoc) || !isLocValid(destLoc)) return srcObj;
+		// if the same location, nothing to do
+		if (srcLoc == destLoc) return srcObj;
+		// if destLoc is child of srcLoc, can't do that move
+		if (isSubLoc(srcLoc, destLoc)) return srcObj;
+
+		var val = getObjectAtLoc(srcObj, srcLoc);
+		var destVal = getObjectAtLoc(srcObj, destLoc);
+		if (val !== undefined) {
+
+			if (destVal === undefined || destVal === null) {
+				// nothing there, add as single element array
+				srcObj = setObjectAtLoc(srcObj, loc, [ val ]);
+			} else if (isArray(destVal)) {
+				// already an array, push new element
+				destVal.push(val);
+			} else if (isObject(destVal)) {
+				// this is an object, do nothing
+			} else {
+				// something else, do nothing
+			}
+			deleteObjectAtLoc(srcObj, srcLoc);
```

Comment:
Should object be deleted if you didn't do the move? Maybe move delete inside the first 2 if clauses.